File: frontend/Assets/Scripts/DataTransfer/Server/AuthSuccessCmd.cs
using System;
using System.IO;
using Attributes;
using Core;
using Utils;


namespace DataTransfer.Server {

    [DTO (DTOCode.AuthSuccess)]
    public class AuthSuccessCmd : ServerCommand {

        public static event Action <AuthSuccessCmd> OnReceived;

        public PlayerInfo PlayerInfo { get; private set; }


        public override void ReadMembers (BinaryReader reader) {
            // todo
            PlayerInfo = new PlayerInfo (reader.ReadInt32 ());
        }


        public override void Execute () {
            The.PlayerInfo = PlayerInfo;
            OnReceived._ (this);
        }

    }

}

File: backend/src/main/java/dto/cmd/server/AuthSuccessCmd.java
package dto.cmd.server;

import dto.DTOCode;
import dto.DTOs;
import io.netty.buffer.ByteBuf;


@DTOCode(DTOs.AUTH_SUCCESS)
public class AuthSuccessCmd extends ServerCommand {

    @Override
    protected void writeMembers (ByteBuf byteBuf) {

    }


    @Override
    protected void readMembers (ByteBuf buffer) {

    }
}
